src/config.js

```diff
@@ -521,6 +524,75 @@ export default class Config {
     }
   }
 
+  async findManifest(dir: string, isRoot: boolean): Promise<?Manifest> {
+    for (const registry of registryNames) {
+      const manifest = await this.tryManifest(dir, registry, isRoot);
+
+      if (manifest) {
+        return manifest;
+      }
+    }
+
+    return null;
+  }
+
+  async findWorktree(initial: string): Promise<?string> {
+    let previous = null;
```

Comment:
Not sure what you mean with `projectManifestJson.workspaces` - if we're not in a workspace (ie. if there is no manifest with a `workspaces` key), then `findWorktree` will return `null`. Do you want it to do something different?

I'll add a check on the experimental flag, tho.